### PR TITLE
Add missing virtual destructors

### PR DESCRIPTION
**Pull request by ivanpauno**
Same idea as https://github.com/ros2/rclcpp/pull/1149.

There's an "overloaded virtual" warning that I'm ignoring in order to avoid breaking API now.
We should fix that in the future.

### CODE REVIEW

File: rviz_rendering/include/rviz_rendering/objects/point_cloud_renderable.hpp
Comment (wjwwood) [inline rviz_rendering/include/rviz_rendering/objects/point_cloud_renderable.hpp]: I can't remember why this was not fixed, but I remember something was complicating the change...

**Review by wjwwood — APPROVED** [review]

**Comment by ivanpauno** [comment]
Jobs config:
* build: ` --packages-up-to rviz2 rviz_default_plugins rviz_rendering rviz_rendering_tests rviz_visual_testing_framework`
* test: ` --packages-select rviz2 rviz_default_plugins rviz_rendering rviz_rendering_tests rviz_visual_testing_framework`

Jobs:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10957/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6298/)
* macOS [Build Status](https://ci.ros2.org/job/ci_osx/8920/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10861/)

File: rviz_rendering/include/rviz_rendering/objects/point_cloud_renderable.hpp
Comment (ivanpauno) [reply rviz_rendering/include/rviz_rendering/objects/point_cloud_renderable.hpp]: It seems actually to be a false positive https://www.ogre3d.org/docs/api/1.7/class_ogre_1_1_simple_renderable.html#a51c5ae135793c602557d46be10c59b8d, as this doesn't seem to be hiding a base class method.

I can delete the TODO.
Besides the false positive, I still prefer adding the new compile option and explicitly ignoring it where it generates a problem.

Comment (wjwwood) [reply rviz_rendering/include/rviz_rendering/objects/point_cloud_renderable.hpp]: Yeah, that's fine, we're already suppressing the clang version of the warning.

Comment (ivanpauno) [reply rviz_rendering/include/rviz_rendering/objects/point_cloud_renderable.hpp]: > Yeah, that's fine, we're already suppressing the clang version of the warning.

I didn't see it before :joy:.
I've deleted the directive for clang, as `GCC diagnostic` works for both https://github.com/ros2/rviz/pull/553/commits/32a232461d2d2e7debabcd39eebb2c226d4c5b98.
I've cleaned up the TODO too.

**Comment by ivanpauno** [comment]
Retriggering linux job to confirm linters are happy:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10958/)

**Comment by ivanpauno** [comment]
Linter failures were fixed and retested on Linux, besides that all the jobs were greenish.
@jacobperron @wjwwood is this gtg?

**Comment by wjwwood** [comment]
I guess so, needs a second review. I just made a release of rviz, but I can roll another.

**Comment by wjwwood** [comment]
I believe this one also does not break API (adding the virtual destructor could be seen as that, but it's a corner case because no one who was calling it before would now have an issue, but I'm open to being convinced otherwise on that one).

**Comment by ivanpauno** [comment]
I don't think it breaks API, but it breaks ABI. So, it can't be backported in the future.

**Comment by wjwwood** [comment]
Yeah, sorry I wasn't suggesting we push it for the patch release, I just was sort of commenting on how the version number would change (also low risk for (re)testing).

**Review by jacobperron — APPROVED** [review]
LGTM

I'm okay with adding this to the initial Foxy release.

**Pull request merged**
